src/node-jspm-jasmine.js

```diff
@@ -3,52 +3,102 @@ import fs from "fs";
 import jspm from "jspm";
 import path from "path";
 import Jasmine from "jasmine";
+import { Instrumenter, Report } from 'istanbul';
+import { remap, writeReport } from "remap-istanbul"
 
+const coveredModules = []
+const tranpiledFileSuffix = '__node-jspm-jasmine-transpiled__.js'
 export function runTests(opts, errCallback = function() {}) {
 
 	let packagePath = '.';
 
 	try {
-
 		if ( typeof opts.packagePath === 'string' ) {
 			require(path.join(process.cwd(), opts.packagePath, 'package.json'));
 			packagePath = path.join(process.cwd(), opts.packagePath);
 		}
-
 	} catch(ex) {
-
 		let errorMessage = ex.toString();
-
 		if (ex.code === 'MODULE_NOT_FOUND' &&
 			errorMessage.indexOf('package.json') > -1) {
 			errorMessage =
 				'Could not find package.json at custom path: ' +
 				path.join(process.cwd(), opts.packagePath, 'package.json');
-	}
-
+		}
 		errCallback(new Error(errorMessage));
 	}
 
 	jspm.setPackagePath(packagePath);
 
 	const jasmine = new Jasmine();
 	const SystemJS = new jspm.Loader();
+	const Instrument = new Instrumenter();
+	const systemInstantiate = SystemJS.instantiate;
+	const coverageFiles = {}
+	const coverageFilePatterns = opts.coverageFiles || [ 'src/**/*.js' ]
+
+	coverageFilePatterns.forEach( pattern => {
+		glob.sync(path.join(process.cwd(),pattern))
+			.forEach(file => coverageFiles[file] = 1)
+	})
+
+	SystemJS.instantiate = (load) => {
+		const normalizedAddress = load.address.replace( 'file://', '' )
+		if (coverageFiles[normalizedAddress]) {
+			load.address = normalizedAddress;
+			coveredModules.push(load.address);
+			const filename = path.basename(load.name);
+			// Start source map processing
+				// all of this could be avoided if we come to a resolution
+				// on this: https://github.com/SitePen/remap-istanbul/issues/48
+				delete load.metadata.sourceMap.sourcesContent;
```

Comment:
There are plenty of people who use JSPM for files that are not ES6 modules (amd modules, umd modules, commonjs modules, typescript, or even just globals etc). Those modules have things that can be tested because they either export things (just not with the `export` keyword) or they expose globals that can be tested. In the case of amd, umd, globals, and commonjs, the files might not be transpiled but SystemJS just provides them a `require` function, a `define` function, etc.

Regarding what we should do in these cases, could we potentially just instrument the file with Istanbul, without having a source map that will be remapped? I'm not sure if remap-istanbul is okay with some files not having source maps, but my guess is that it would be fine??